session04/mailroomDictFile.py

```diff
@@ -0,0 +1,126 @@
+and creating a txt file for each letter with donor's name as file's name
+"""
+
+# creating global list so it can be accessed from different functions
+# donor_list is not a diction
+donor_dict = {'Albert White': [1254.45, 8966.70],
+              'Susan Smith': [9856.94, 1245.78],
+              'Daniel Motoko': [5897.85, 20000.48],
+              'Alison Dempsey': [2565.44, 500.32],
+              'Patrick Johnson': [6578.55, 4879.23, 5879.99]}
+
+
+def menu():
+    """ print menu for user to select from """
+    question = input("\nPlease select options 1, 2, or 3, 4 from the menu "
+                         "or 'list' to view donor names: \n"
+                         "1. Send a Thank you letter\n"
+                         "2. Create a Report\n"
+                         "3. Send letters to everyone\n"
+                         "4. Quit\n"
+                         "What's your choice?\n")
+    return question
+
+def print_donor_list():
+    """ printing 0th element from dict (printing only donors' name)"""
+    print("Donor Name:\n")
+    for name in donor_dict:
+        print(name)
+
+def find_donor(name):
+    """ finding donor's name in donor_dict and printing their information """
+    for donor in donor_dict:
+        # case-insenstive compare
+        if name.lower() == donor.lower():
+            return donor
+    return None
+
+def sort_key(item):
+    """ key function used to sort the list by first (not zeroth) item"""
+    return item[1]
+
+
+def print_letter(name, amount):
+    """ writing a thank you letter to donor """
+    print("\nDear {},\n\nThank you for your kind donation of ${}."
+          "\n\nSincerely,\n\nA Team".format(name, amount))
+
+# def letters_for_everyone():
+#     """ writing a thank you letter to all donor """
+#     for name, amounts in donor_list.items():
+#         print("\nDear {},\n\nThank you for your kind donation of ${}."
+#           "\n\nSincerely,\n\nA Team".format(name, amounts))
+
+def letters_for_everyone():    #need to go back and put letters in txt file.
+    """ writing a thank you letter to all donor """
+    for name, amounts in donor_dict.items():
+        new_amounts = []
+        for i in amounts:
+            new_amounts.append('$' + str(i))
+        all_amounts = ", ".join(new_amounts)
+        with open(name + ".txt", 'w') as this_file:
+            this_file.write("\nDear {},\n\nThank you for your kind donation of ${} ({})."
+                  "\n\nSincerely,\n\nA Team".format(name, round(sum(amounts), 2), all_amounts))
+
+
+def print_report():
+    """ printing a report that includes donor names, contribution amounts, sum, and average """
+    report = []
+    for (name, amount) in donor_dict.items():
+        total_amount = sum(amount)
+        number_gifts = len(amount)
+        avg_gift = total_amount / number_gifts
+        report.append((name, total_amount, number_gifts, avg_gift))
+
+    # sorting report data A-Z
+    report.sort(key=sort_key)
```

Comment:
It would be a good idea to split print_report into two functions: something like generate_report that returns the report variable, and the print stuff below this comment, that just prints stuff. That way you can write a test that makes sure the data in report is correct, without worrying about the report format.